sklearn/utils/testing.py

```diff
@@ -422,6 +424,70 @@ def assert_raise_message(exceptions, message, function, *args, **kwargs):
                              (names, function.__name__))
 
 
+def _assert_same_model_method(method, X, estimator1, estimator2, msg=None):
+    if hasattr(estimator1, method):
+        m = '%r\n\nhas %s, but\n\n%r\n\ndoes not' % (estimator1,
+                                                     method,
+                                                     estimator2)
+
+        if not hasattr(estimator2, method):
+            raise AttributeError(m)
+
+        # Check if the method(X) returns the same for both models.
+        res1 = getattr(estimator1, method)(X)
+        res2 = getattr(estimator2, method)(X)
+        same_model = (res1.shape == res2.shape) and np.allclose(res1, res2)
+
+        if msg is None:
+            msg = ("Models are not equal. \n\n%s method returned"
+                   " different results:\n\n%r\n\n and\n\n%r\n\n for :\n\n%r"
+                   "\n\nand :\n\n%r" % (method, res1, res2,
+                                        estimator1, estimator2))
+
+        assert same_model, msg
+
+
+def assert_same_model(X, estimator1, estimator2, msg=None):
+    """Helper function to check if models are same"""
+    _assert_same_model_method('predict', X, estimator1, estimator2, msg)
+    _assert_same_model_method('transform', X, estimator1, estimator2, msg)
+    _assert_same_model_method('decision_function',
+                              X, estimator1, estimator2, msg)
+    _assert_same_model_method('predict_proba',
+                              X, estimator1, estimator2, msg)
+    assert_fitted_attributes_equal(estimator1, estimator2)
+
+
+def assert_not_same_model(X, estimator1, estimator2, msg=None):
+    """Helper function to check if models are different"""
```

Comment:
Note that it should be used with caution because it produces false negatives (i.e. `AssertionError` will be raised when models differ but our means of assessment is not powerful enough).